rayon-core/src/future/mod.rs

```diff
@@ -82,6 +82,11 @@ pub fn new_rayon_future<'scope, F, S>(future: F, scope: S) -> RayonFuture<F::Ite
     }
 }
 
+/// Implements a wait method which should not cause race conditions.
+/// Caution should still be used since it is not the most optimum resource
+/// usage.  It is typically better to create and spawn a new future.
+/// Recommend using:
+///   scope.spawn_future(future.and_then(|v| use(v))
```

Comment:
Here the formatting doesn't quite seem right. Also, the example should be something that can actually execute, since our tests are actively tested by rustdoc. Perhaps something like this (untested):

    It is typically better to create and spawn a new future using `and_then`, like so:
    
    ```rust
    extern crate futures;
    scope(|s| {
        let future = s.spawn(futures::lazy(|| -> Result<usize, ()> {
            // insert expensive computation here
            Ok(22)
        });

        // instead of `let x = future.rayon_wait()`, try this:
        let future = s.spawn(future.and_then(|x| -> Result<usize, ()> {
            // code that executes after `future` is complete;
            // `x` is the result value
            Ok(x * 2)
        });
    ```
```